### ISSUE TITLE
DataAnnotations: Inverse Property is not generated

### ISSUE THREAD

**Issue by inforithmics** [ReversePOCO/EntityFramework-Reverse-POCO-Code-First-Generator#340]
```
public partial class Task
{
    ...
     public string SearchCriteria1 { get; set; }
    ...
    [ForeignKey("SearchCriteria1")] public virtual Suchkriterium SearchCriteriaEntity1 { get; set; } 
    ....
}
public partial class SearchCriteria
{
   ...
   [InverseProperty("SearchCriteriaEntity1")] 
   public virtual System.Collections.Generic.ICollection<Task> Task1 { get; set; } 
   ...
}
```

Above is a pseudoCode Example of InverseProperty the String value is the name of the Property that holds the Entity of Type SearchCriteriaEntity1 in this case. 

Normally it isn't necessary to specify it, but if more than two Collection Properties with the Same Type exist it is necessary to specify or else strange bugs occur.

Currently I use ForeignKeyAnnotationsProcessing to Specify the InverseProperty Attributes.

**Comment by inforithmics** [comment]
I made following change to the EF.Reverse.POCO.Core.ttinclude

Changed Signature of ForeignKeyAnnotationsProcessing
from
Func<Table, Table, string, string[]> ForeignKeyAnnotationsProcessing;
to
Func<ForeignKey, Table, Table, string, string[]> ForeignKeyAnnotationsProcessing;

fixed all compilation errors

and add following code into ForeignKeyAnnotationsProcessing in EF.Reverse.POCO.tt

```
        var fkColumn = fkTable.Columns.FirstOrDefault(f => f.Name == foreignKey.FkColumn);
        var pkColumn = pkTable.Columns.FirstOrDefault(f => f.Name == foreignKey.PkColumn);

        if (pkColumn != null && pkColumn.IsPrimaryKey && fkColumn.EntityFk.Count > 0)
        {
            var entityFk = fkColumn.EntityFk[0];
            var definition = entityFk.Definition;
            var curlyBraketPos = definition.IndexOf("{");
            if (curlyBraketPos > 0)
            {
                var propertyNameStart = definition.LastIndexOf(" ", curlyBraketPos - 2);
                if (propertyNameStart > 0)
                {                    
                    var entityProperty = definition.Substring(propertyNameStart, curlyBraketPos - propertyNameStart);
                    entityProperty = entityProperty.Trim();
                    return new[] {"InverseProperty(\"" + entityProperty + "\")"};
                }
            }
        }
```

The Code is a bit hacky because in the Column.EntityFk I have to parse the ColumnName out of the Definition a little more semantic would be useful here.

**Comment by sjh37** [comment]
Released v2.37.2
